tests/store_test.go

```diff
@@ -40,7 +40,7 @@ type StoreSuite struct {
 func (s *StoreSuite) TestStoreNew() {
 	doc := NewStoreWithConstructFixture("foo")
 	s.False(doc.IsPersisted())
-	s.False(doc.ID.IsEmpty())
+	s.True(doc.ID.IsEmpty())
```

Comment:
Yes, but it is not added in the `New{foo}` method, but when you are going to insert just as a last resort.

If you need to have an ID, it's your job to provide one using the private constructor. That should be in the docs, cause it can be misleading.